### PR TITLE
scrape: catch errors when creating HTTP clients

### LINKED ISSUE #5172: SIGSEGV Segmentation fault while scrape deleted kubernetes endpoints
## Bug Report
**What did you do?**
We had some config to monitor some kubernetes endpoint but when some files and resources are deleted (e.g. certificate files) prometheus crashes.

**What did you expect to see?**
Capture and return errors during scrapping preventing prometheus crash.

**What did you see instead? Under which circumstances?**
When deleting kubernetes cluster and all files associated to that cluster.

**Environment**
* System information:
linux 4.19.9-coreos

* Prometheus version:
prometheus version=2.6.0

* Prometheus configuration file:
```
- job_name: guest-cluster-2sd23-apiserver
  scheme: https
  kubernetes_sd_configs:
  - api_server: https://master.2sd23
    role: endpoints
    tls_config:
      ca_file: /certs/2sd23-ca.pem
      cert_file: /certs/2sd23-crt.pem
      key_file: /certs/2sd23-key.pem
      insecure_skip_verify: false
```

* Logs:
```
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x6675c2]

goroutine 2529 [running]:
net/http.(*Client).deadline(0x0, 0xc019b57208, 0x40bb2f, 0xc015cd40e0)
        /usr/local/go/src/net/http/client.go:187 +0x22
net/http.(*Client).do(0x0, 0xc04b85ef00, 0x0, 0x0, 0x0)
        /usr/local/go/src/net/http/client.go:527 +0xab
net/http.(*Client).Do(0x0, 0xc04b85ef00, 0x23, 0xc02580e4c0, 0x9)
        /usr/local/go/src/net/http/client.go:509 +0x35
github.com/prometheus/prometheus/scrape.(*targetScraper).scrape(0xc0203c1f50, 0x1eb97e0, 0xc00a3c7a40, 0x1e997a0, 0xc0515e1b20, 0x0, 0x0, 0x0, 0x0)
        /app/scrape/scrape.go:471 +0x111
github.com/prometheus/prometheus/scrape.(*scrapeLoop).run(0xc017ed6e80, 0xdf8475800, 0xdf8475800, 0x0)
        /app/scrape/scrape.go:813 +0x487
created by github.com/prometheus/prometheus/scrape.(*scrapePool).sync
        /app/scrape/scrape.go:336 +0x45d
```

### PR DESCRIPTION
**Pull request by simonpasquier**
This PR adds 4 new (counter) metrics:
* `prometheus_target_scrape_pools_total`
* `prometheus_target_scrape_pools_failed_total`
* `prometheus_target_scrape_pool_reloads_total`
* `prometheus_target_scrape_pool_reloads_failed_total`

Closes #5172 

The other option would be to enforce the validity of the TLS configuration when unmarshaling the configuration but that would break `promtool check config`.

### CODE REVIEW

File: scrape/manager.go [hunk 1]
```diff
@@ -104,26 +105,27 @@ func (m *Manager) reload() {
 	m.mtxScrape.Lock()
 	var wg sync.WaitGroup
 	for setName, groups := range m.targetSets {
-		var sp *scrapePool
-		existing, ok := m.scrapePools[setName]
+		_, ok := m.scrapePools[setName]
```
Comment (tariq1890) [inline scrape/manager.go]: How about this ?

```suggestion
		if _, ok := m.scrapePools[setName]; ok! {
```

File: scrape/scrape.go [hunk 2]
```diff
@@ -61,6 +62,32 @@ var (
...
+			Name: "prometheus_target_scrape_pools_failed_total",
+			Help: "Total number of scrape pool creations that failed.",
+		},
+	)
+	targetScrapePoolReloads = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "prometheus_target_scrape_pool_reloads_total",
+			Help: "Total number of reloads on a scrape pool.",
+		},
+		[]string{"scrape_job"},
```
Comment (brian-brazil) [inline scrape/scrape.go]: There could be a lot of scrape jobs, I'd not have this label.

Comment (simonpasquier) [reply scrape/scrape.go]: Fine for me. Note that `prometheus_target_sync_length_seconds` and `prometheus_target_scrape_pool_sync_total` already include that label.

**Comment by brian-brazil** [comment]
:+1:

**Pull request merged**